pkg/controller/user/user_controller.go

```diff
@@ -531,6 +570,74 @@ func (c *Controller) deleteRoleBindings(user *iamv1alpha2.User) error {
 	return nil
 }
 
+func (c *Controller) deleteLoginRecords(user *iamv1alpha2.User) error {
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}).String(),
+	}
+	deleteOptions := metav1.NewDeleteOptions(0)
+
+	if err := c.ksClient.IamV1alpha2().LoginRecords().
+		DeleteCollection(deleteOptions, listOptions); err != nil {
+		klog.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) syncUserStatus(user *iamv1alpha2.User) (*iamv1alpha2.User, error) {
+	records, err := c.loginRecordLister.List(labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}))
```

Comment:
If I was you, I would put the following at the beginning. That could save us listing login records
```
if user == nil ||  (user.Status != nil && user.Status.State == iamv1alpha2.UserDisabled) {
  return user, nil
}
```